apps/glusterfs/operations_manage.go

```diff
@@ -0,0 +1,209 @@
+//
+// Copyright (c) 2018 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package glusterfs
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/heketi/heketi/executors"
+)
+
+// OpCounter is used to track and manage how many operations are being
+// processed by the server.
+type OpCounter struct {
```

Comment:
Couple of questions on locking: The big lock is taken for each operation or start of each request , Isnt it ? second question is, how the lock is handled at server restart or respawn?

Reply:
A read lock is taken to check the current number of in-flight operations but is released when the function returns. Similarly, write locks are taken when the increment or decrement function is called but only long enough to change the counter value. The lock is not held the majority of the time when the operation is being processed. The lock is always unlocked when the struct is initialized, this is the default of the golang lock construct.